geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GeneralShapefileIngest.scala

```diff
@@ -43,56 +35,45 @@ object GeneralShapefileIngest {
     val shapefile = getShapefileDatastore(shapefilePath)
-    val features = shapefile.getFeatureSource.getFeatures
-    val newDS = featuresToDataStore(features, ds, featureName)
-    shapefile.dispose()
-    newDS
-  }
-
-  def featuresToDataStoreViaParams(features: SimpleFeatureCollection,
-                                   params: JMap[String, Serializable],
-                                   featureName: String): DataStore = {
-    val ds = DataStoreFinder.getDataStore(params)
-    featuresToDataStore(features, ds, featureName)
+    try {
+      ingestToDataStore(shapefile.getFeatureSource.getFeatures, ds, typeName)
+    } finally {
+      if (shapefile != null) {
+        shapefile.dispose()
+      }
+    }
   }
 
-  def featuresToDataStore(features: SimpleFeatureCollection,
-                          ds: DataStore,
-                          featureName: String): DataStore = {
+  def ingestToDataStore(features: SimpleFeatureCollection, ds: DataStore, typeName: Option[String]): Unit = {
     // Add the ability to rename this FT
-    val featureType: SimpleFeatureType =
-      if(featureName != null) {   // Is this line right?
-        val originalFeatureType = features.getSchema
-        val sftBuilder = new SimpleFeatureTypeBuilder()
-        sftBuilder.init(originalFeatureType)
-        sftBuilder.setName(featureName)
-        sftBuilder.buildFeatureType()
-      } else
-        features.getSchema
+    val featureType: SimpleFeatureType = typeName.map { name =>
+      val sftBuilder = new SimpleFeatureTypeBuilder()
+      sftBuilder.init(features.getSchema)
+      sftBuilder.setName(name)
+      sftBuilder.buildFeatureType()
+    }.getOrElse(features.getSchema)
 
     val featureTypeName = featureType.getName.getLocalPart
 
-    val existingSchema = Try { ds.getSchema(featureTypeName) }.getOrElse(null)
+    val existingSchema = Try(ds.getSchema(featureTypeName)).getOrElse(null)
     if (existingSchema == null) {
       ds.createSchema(featureType)
     }
 
-    val newType = ds.getSchema(featureTypeName)
-
-    val reTypedSFC = new TypeUpdatingFeatureCollection(features, newType)
-
-    val fs: FeatureStore[SimpleFeatureType, SimpleFeature] =
-      ds.getFeatureSource(featureTypeName).asInstanceOf[FeatureStore[SimpleFeatureType, SimpleFeature]]
-    val transaction = new DefaultTransaction("create")
-    fs.setTransaction(transaction)
-
-    fs.addFeatures(reTypedSFC)
-    transaction.commit()
-    transaction.close()
-
-    ds
+    val reTypedSFC = new TypeUpdatingFeatureCollection(features, ds.getSchema(featureTypeName))
+
+    WithClose(ds.getFeatureWriterAppend(featureTypeName, Transaction.AUTO_COMMIT)) { writer =>
+      SelfClosingIterator(reTypedSFC.features()).foreach { feature =>
+        try {
+          FeatureUtils.copyToWriter(writer, feature)
+          writer.write()
+        } catch {
+          case NonFatal(e) => logger.warn(s"Error writing feature: $feature: $e", e)
```

Comment:
Nice!  Did you have a chance to try this with the provided shapefile?  How many features fail?